client/packages/service-message/query/use-service-message.ts

```diff
@@ -0,0 +1,59 @@
+
+import { useContext, useEffect, useState } from "react";
+import { AppServiceMessage, ServiceMessageContext } from "../provider/ServiceMessageProvider";
+
+import { ServiceMessage } from "../types/types";
+
+export const useServiceMessage = (appKey?: string) => {
+  const context = useContext(ServiceMessageContext);
+
+  if (!context) {
+    throw new Error("ServiceMessageContext context used out of bounds");
+  }
+
+  useEffect(() => {
+    if (appKey) context.serviceMessages.setCurrentApp(appKey)
+  }, [appKey])
+
+  const [messages, setMessages] = useState<ServiceMessage[]>([]);
+  const [appsMessages, setAppsMessages] = useState<AppServiceMessage[]>([]);
+  const [portalMessages, setPortalMessages] = useState<ServiceMessage[]>([]);
+  const [currentMessages, setCurrentMessages] = useState<ServiceMessage[]>([]);
+
+  useEffect(() => {
+    const sub = context.serviceMessages.messages$.subscribe(setMessages);
+    return () => {
+      sub.unsubscribe()
+    }
+  }, []);
```

Comment:
Should probably have context in the dependency array

Reply:
ill add it but i dont see the need for it